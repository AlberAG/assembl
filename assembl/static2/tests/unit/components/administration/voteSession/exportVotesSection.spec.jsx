// @flow
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';

import ExportVotesSection from '../../../../../js/app/components/administration/voteSession/exportVotesSection';

configure({ adapter: new Adapter() });

describe('ExportVotesSection component', () => {
  const props = {
    debateId: 'my-debate',
    voteSessionId: 'my-vote-session'
  };

  it('should render an export section for votes', () => {
    const component = renderer.create(<ExportVotesSection {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should set exportRoute state when user clicks on radio button', () => {
    const wrapper = mount(<ExportVotesSection {...props} />);
    const options = wrapper.find('input');
    expect(wrapper.state('exportRoute')).toEqual('vote_results_csv');
    const eventMock = {
      target: {
        value: 'extract_csv_voters'
      }
    };
    options.last().simulate('change', eventMock);
    expect(wrapper.state('exportRoute')).toEqual('extract_csv_voters');
    eventMock.target.value = 'vote_results_csv';
    options.first().simulate('change', eventMock);
    expect(wrapper.state('exportRoute')).toEqual('vote_results_csv');
  });

  describe('getExportLink method', () => {
    it('should return a link to export votes', () => {
      const wrapper = shallow(<ExportVotesSection {...props} />);
      const actual = wrapper.instance().getExportLink();
      const expected = '/data/Discussion/my-debate/widgets/my-vote-session/vote_results_csv';
      expect(actual).toEqual(expected);
    });

    it('should return a link to export csv voters', () => {
      const wrapper = shallow(<ExportVotesSection {...props} />);
      wrapper.instance().setState({ exportRoute: 'extract_csv_voters' });
      const actual = wrapper.instance().getExportLink();
      const expected = '/data/Discussion/my-debate/widgets/my-vote-session/extract_csv_voters';
      expect(actual).toEqual(expected);
    });
  });
});