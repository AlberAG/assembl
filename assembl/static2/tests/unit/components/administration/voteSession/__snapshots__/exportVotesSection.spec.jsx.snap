// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportVotesSection component should render an export section for votes 1`] = `
<div
  className="admin-box admin-export-section"
>
  <div>
    <h3
      className="dark-title-3"
    >
      Export data
    </h3>
    <div
      className="box-hyphen"
    />
    <div
      className="annotation"
    >
      You can export all the vote session data by clicking on the Export button
    </div>
  </div>
  <div
    className="admin-content"
  >
    <div
      className="form-group"
    >
      <div
        className="radio"
        style={undefined}
      >
        <label
          title=""
        >
          <input
            checked={true}
            disabled={false}
            name="exportLink"
            onChange={[Function]}
            type="radio"
            value="vote_results_csv"
          />
          <span
            className={undefined}
            style={undefined}
          >
            Export general data for the vote module
          </span>
        </label>
      </div>
      <div
        className="radio"
        style={undefined}
      >
        <label
          title=""
        >
          <input
            checked={false}
            disabled={false}
            name="exportLink"
            onChange={[Function]}
            type="radio"
            value="extract_csv_voters"
          />
          <span
            className={undefined}
            style={undefined}
          >
            Export vote details for each user
          </span>
        </label>
      </div>
    </div>
    <br />
    <a
      className="button-link button-dark margin-l"
      href="/data/Discussion/my-debate/widgets/my-vote-session/vote_results_csv"
      onClick={[Function]}
      style={Object {}}
    >
      <span
        className={undefined}
        style={undefined}
      >
        Export
      </span>
    </a>
  </div>
</div>
`;
